GitUI/CommandsDialogs/FormRemotes.cs

```diff
@@ -668,5 +672,95 @@ private void ShowSeparatePushUrl(bool visible)
                 ? _labelUrlAsFetch.Text
                 : _labelUrlAsFetchPush.Text;
         }
+
+        private void Url_Enter(object sender, EventArgs e)
+        {
+            FillWithSomeGeneratedRemoteUrls(Url, r => r.Url);
+        }
+
+        private void ComboBoxPushUrl_Enter(object sender, EventArgs e)
+        {
+            FillWithSomeGeneratedRemoteUrls(comboBoxPushUrl, r => r.PushUrl);
+        }
+
+        private void FillWithSomeGeneratedRemoteUrls(CaseSensitiveComboBox combobox, Func<ConfigFileRemote, string> urlGetter)
+        {
+            if (RemoteName.Text is "origin" or "upstream")
```

Comment:
❕ These should be configurable. For example, I never use "upstream", and instead use "fork", "remote", "internal", etc.

Reply:
Done but here it's not a big deal if a term is not known because the user will just have an url proposed that it will have to edit.
I have still changed a little the behavior in this case so that url are still added to the list in this case (because it still can be useful) but not automatically fill the selected value.